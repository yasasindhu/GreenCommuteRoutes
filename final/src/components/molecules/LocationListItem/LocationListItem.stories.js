import React from "react";

import LocationListItem from "./LocationListItem";

export default {
    title: "molecules/LocationDisplay",
    component: LocationListItem,
};
const Template = (args) => <LocationListItem {...args} />;

export const PrimaryLocation = Template.bind({});


PrimaryLocation.args = {
    locationName : 'East Marredpally, Secunderabad'
};