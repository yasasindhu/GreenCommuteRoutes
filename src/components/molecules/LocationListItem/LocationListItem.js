import React from "react";
import {
  ListItem,
  ListItemIcon,
  Typography,
} from "@material-ui/core";
import { makeStyles, ThemeProvider } from '@material-ui/core/styles';
import customtheme from "../../theme";
import { RoomOutlined } from "@material-ui/icons";

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'inline-flex',
    flexDirection: 'row',
    alignItems: 'center',

  },
  typoStyle: {
    textOverflow: 'ellipsis',
    overflow: 'hidden'
  },
  outlinedIcon: {
    padding: "0px 0px 0px 20px",
    width: '24px', height: '24px', color: '#9bbdcb'
  },
}));


export default function LocationListItem({ locationName }) {

  const classes = useStyles();

  return (
    <ThemeProvider theme={customtheme}>
      <div className={classes.root} data-testid="divinlocationlistitem">
        <ListItem data-testid="lisiteminlocationlistitem">
          <ListItemIcon data-testid="listitemiconinlistitem">
            <RoomOutlined className={classes.outlinedIcon} data-testid="roominlocationlistitem" />
          </ListItemIcon>
          <Typography variant="subtitle1" className={classes.typoStyle} data-testid="locationnameinlocationlistitem">{locationName}</Typography>
        </ListItem>
      </div>
    </ThemeProvider>
  );
}