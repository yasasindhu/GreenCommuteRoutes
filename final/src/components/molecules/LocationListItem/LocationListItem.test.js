import { render, screen, cleanup } from '@testing-library/react';
import LocationListItem from './LocationListItem';



describe('Typography Rendered', () => {
   
    test('check the Descption Card title', () => {
      const { asFragment} = render(<LocationListItem locationName="hyd"/>);
        expect(asFragment()).toMatchSnapshot();
    });
  });