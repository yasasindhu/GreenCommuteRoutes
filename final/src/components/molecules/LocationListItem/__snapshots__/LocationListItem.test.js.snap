// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography Rendered check the Descption Card title 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-1"
    data-testid="divinlocationlistitem"
  >
    <li
      class="MuiListItem-root MuiListItem-gutters"
      data-testid="lisiteminlocationlistitem"
    >
      <div
        class="MuiListItemIcon-root"
        data-testid="listitemiconinlistitem"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-outlinedIcon-3"
          data-testid="roominlocationlistitem"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zM7 9c0-2.76 2.24-5 5-5s5 2.24 5 5c0 2.88-2.88 7.19-5 9.88C9.92 16.21 7 11.85 7 9z"
          />
          <circle
            cx="12"
            cy="9"
            r="2.5"
          />
        </svg>
      </div>
      <h6
        class="MuiTypography-root makeStyles-typoStyle-2 MuiTypography-subtitle1"
        data-testid="locationnameinlocationlistitem"
      >
        hyd
      </h6>
    </li>
  </div>
</DocumentFragment>
`;
